en/docs/visual-programming/data-mapper.md

```diff
@@ -1 +1,160 @@
+
+![Fix Errors](img/dm-fix-diagnostics.gif "Fix incompatible types error")
+
+Once you fix the error, the connection appears in blue to indicate that there are no errors.
+
+### Aggregate multiple input fields to one output field
+You can derive one output parameter by combining two or more input parameters. In this example, the value for the fullName output parameter is a combination the values for the firstName and lastName input parameters. You can map them as shown below:
+>**Info:** To aggregate fields, you can map two or more fields to the same output field. The Data Mapper will automatically combine the two fields and assign it to the output field. By default, the fields will be combined with a plus operator. If you want to use a different operator or method to combine two fields, you can click on the code button and customize the expression with the expression editor.
+
+![Concatenate](img/dm-concatination.gif "Aggregate multiple input fields")
+
+### Map the arrays
+To convert from one array type to another, you can simply map the input array to the output array. If the arrays are compatible, they will be connected with a blue line. If they are not compatible, the connecting line will appear in red.
+
+>**Info:** You can use Ballerina query support to convert one array type to another. To use a query in a Data Mapper, you can select the array by clicking on it. Then, it will provide you with several buttons. Click the code action button (bulb sign) and select **Convert to query**. Then, the Data Mapper will convert the mapping to a query. Then, move into the query and do the mapping between the array types.
+
+![Convert to Query](img/dm-convert-to-query.gif "Map via query expression")
+
+#### Further process the data
+You can further process the data within the query expression. Currently, the Data Mapper supports the following intermediate clauses.
+- Where clause: filter data based on a given condition
+- Let clause: define local variables within the query expression
+- Limit clause: limit the number of elements returns from the query expression
+- Order by clause: sort data within the query expression in `ascending` or `descending` directions
+- Join clause: performs an inner or left outer join
+- Outer join clause: performs an outer join
```

Comment:
```suggestion
- `Outer join` clause: performs an outer join
```
Should we define the `Inner join` clause also?

Reply:
The `join clause` is actually performing the inner join, Fixed with 0f05ec1c9c116c20045ee1b7ec3e5bdf24ebcc92